elements/lisk-p2p/src/peer/base.ts

```diff
@@ -492,17 +496,28 @@ export class Peer extends EventEmitter {
 			this._productivity = { ...DEFAULT_PRODUCTIVITY };
 		}
 	}
-
 	private _updateFromProtocolPeerInfo(rawPeerInfo: unknown): void {
+		if (!this._serverNodeInfo) {
+			throw new Error('Missing server node info.');
+		}
+
 		const protocolPeerInfo = {
 			...rawPeerInfo,
 			ip: this._ipAddress,
 			wsPort: this._wsPort,
 		};
+
 		const newPeerInfo = validatePeerInfo(
 			protocolPeerInfo,
 			this._peerConfig.maxPeerInfoSize,
 		);
+
+		const result = validatePeerCompatibility(newPeerInfo, this._serverNodeInfo);
+
+		if (!result.success && result.error) {
+			throw new Error(result.error);
```

Comment:
we can throw a similar error to `PeerInboundHandshakeError` at least with peer ip and port for logging purpose